query-dsl/src/main/java/org/infinispan/query/dsl/continuous/ContinuousQuery.java

```diff
@@ -0,0 +1,23 @@
+package org.infinispan.query.dsl.continuous;
+
+import org.infinispan.query.dsl.Query;
+
+import java.util.List;
+
+/**
+ * A container of continuous query listeners for a cache.
+ * <p>Implementations are not expected to be threadsafe.
+ *
+ * @author anistor@redhat.com
+ * @since 8.2
+ */
+public interface ContinuousQuery<K, V> {
+
+   <C> void addContinuousQueryListener(Query query, ContinuousQueryListener<K, C> listener);
+
+   void removeContinuousQueryListener(ContinuousQueryListener<K, ?> listener);
+
+   List<ContinuousQueryListener<K, ?>> getListeners();
+
+   void removeAllListeners();
+}
```

Comment:
They are not really part of the DSL. I'll move them to ```org.infinispan.query.continuous``` package. I think we could rename ```query-dsl``` module to ```query-api``` or something like that.